### ISSUE TITLE
Ignore the "." (period) in street names during voice output

### ISSUE THREAD

**Issue by MarkuBu** [osmandapp/OsmAnd#18633]
### Description

If street names have a "." the voice output is paused at this point for a second. This interferes with navigation.

Examples from my personal experience:
St. Kathrin
Dr. Dietz Straße

### Steps to reproduce

Just navigate through a street with a "." in the name.

### Actual result

The actual output is like:

Turn right to Sankt                          Kathrin
Turn left to Doktor                          Dietz Straße

### Expected result

Turn right to Sankt Kathrin
Turn left to Doktor Dietz Straße

### Your Environment (required)

```
OsmAnd Version: 4.5.10
Android/iOS version: Android 14
Device model: Pixel 7
```

**Comment by MarkuBu** [comment]
That's exactly the problem, except "St." doesn't mean "street" in Germany.

**Comment by DmitryAlexei** [comment]
Oh yes! St. is also voiced incorrectly as street, not Saint.

**Comment by vshcherb** [comment]
Still removing dot might cause other issues, so only if we do correct spelling then it's fixed St. -> Sant / Saint

**Comment by sonora** [comment]
Yes, I am speaking from experience when I say you have to be _very careful_. Even if you can reproduce an issue on some particular devices and setups, many of these issues are extremely sensitive to the TTS engine and associated download data used, the device's regional settings, OS version, etc.

Many of these things only occur in certain circumstances, while in others and/or in other configurations they are handled by the TTS engine perfectly correct. Oftentimes, the handling of these situations also gets corrected as TTS engines progress, ie.e disappear over time. So if we decide to do something about it in our code
- any exception we define should be at least region/language-aware
- should be tested on a broad range of devices and localizations/configurations
- may have to be maintained over time (but often is not)
- and if in doubt, it is often better to not create such an exception.

We have a limited number of exceptions [here](https://github.com/osmandapp/OsmAnd/blob/8343aa375d66a167bce31e0fdc293072e907a190/OsmAnd/src/net/osmand/plus/routing/VoiceRouter.java#L641), most of which were likely introduced by me many years ago, and some are probably outdated by now and should be removed, but nobody has the impulse and time to do and test that... ;)

I advise to really stay with @vshcherb  's judgement here "Nice to have" and wait this out. Quite possibly, your mileage may vary, but if e.g. you update your TTS engine, or try changing it e.g. test Samsung's instead of Google's, that may already resolve many of these issues.
